### PR TITLE
Add Security and UX sections to the canonical RFD

### PR DESCRIPTION
**Pull request by codingllama**
Add Security and UX sections to the canonical RFD as a reminder of the importance of those topics.

### CODE REVIEW

**Comment by codingllama** [comment]
FYI @zmb3 @russjones.

Added sections under Details, as it seemed appropriate.

I've kept instructions rather vague in an attempt to encourage a more free-form discussion of the new sections, let me know if you have other ideas.

**Review by zmb3 — APPROVED** [review]

**Review by klizhentas — APPROVED** [review]

File: rfd/0000-rfds.md [hunk 1]
```diff
@@ -101,3 +101,11 @@ The purpose of the `state` is to tell the reader whether they should care about
 this RFD at all. For example, `deprecated` RFDs can be skipped most of the
 time. `implemented` is relevant to Teleport users, but `draft` is mostly for
 Teleport engineers and early adopters.
+
+### Security
+
+Describe the security considerations for your design doc.
```
Comment (klizhentas) [inline rfd/0000-rfds.md:107]: @codingllama would be helpful to give some guidance here:

* Explore possible attack vectors and see how Teleport protects them
* If design if frontend, explore common web vulnerabilities and see if design would prevent them
* Explore possible DDOS and other outage type attacks and see how design considers protecting from them
* If using crypto, show that best practices were considered

File: rfd/0000-rfds.md [hunk 2]
```diff
@@ -101,3 +101,11 @@ The purpose of the `state` is to tell the reader whether they should care about
 this RFD at all. For example, `deprecated` RFDs can be skipped most of the
 time. `implemented` is relevant to Teleport users, but `draft` is mostly for
 Teleport engineers and early adopters.
+
+### Security
```
Comment (klizhentas) [inline rfd/0000-rfds.md:105]: @codingllama would be helpful to give some guidance here:

* Explore possible attack vectors and see how Teleport protects them
* If design if frontend, explore common web vulnerabilities and see if design would prevent them
* Explore possible DDOS and other outage type attacks and see how design considers protecting from them
* If using crypto, show that best practices were used to define new crypto
* Same for authentication and authorization, if any new type of auth / authz introduced, explain the design

File: rfd/0000-rfds.md [hunk 3]
```diff
@@ -101,3 +101,11 @@ The purpose of the `state` is to tell the reader whether they should care about
...
+
+Describe the security considerations for your design doc.
+
+### UX
+
+Describe the UX changes and impact of your design doc.
```
Comment (klizhentas) [inline rfd/0000-rfds.md:111]: Let's give some guidance here as well:

Explore UI, CLI and API user experience by diving through common scenarios that users would go through. 
When exploring, show CLI, UI and API requests and responses, users would observe
Consider making error messages actionable, explore common type of failure modes and how users can recover

**Review by klizhentas — CHANGES_REQUESTED** [review]
Added a couple of notes, let's give folks some guidance on those sections

File: rfd/0000-rfds.md (hunk 2 above)
Comment (codingllama) [reply rfd/0000-rfds.md:105]: Done.

File: rfd/0000-rfds.md (hunk 3 above)
Comment (codingllama) [reply rfd/0000-rfds.md:120]: Done.

**Comment by codingllama** [comment]
Thanks for the edits, @klizhentas. I've added all of your suggestions plus a few of my own.

PTAL?

**Review by klizhentas — APPROVED** [review]

**Pull request merged**